frodo.js

```diff
@@ -1,83 +1,80 @@
 /**
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?